llpc/context/llpcCompiler.cpp

```diff
@@ -1155,6 +1192,28 @@ Result Compiler::buildPipelineInternal(Context *context, ArrayRef<const Pipeline
   return result;
 }
 
+// =====================================================================================================================
+template <typename CacheResult, typename UserCache, typename Cache, typename HashId, typename EntryHandle>
+void findCaches(unsigned *stage, UserCache *cache, Cache **outFragCache, Cache **outNonFragCache,
+                CacheResult *fragCacheResult, CacheResult *nonFragCacheResult, CacheResult cmpResult, HashId *fragHash,
+                HashId *nonFragHash, BinaryData *fragElf, BinaryData *nonFragElf, EntryHandle *fragEntry,
+                EntryHandle *nonFragEntry,
+                std::function<CacheResult(UserCache *, HashId *, BinaryData *, Cache **, EntryHandle *)> lookupCB) {
```

Comment:
Since `findCaches` is already a template function, `lookupCB` could also be a template parameter to avoid the `std::function` overhead.